### ISSUE TITLE
Running single requests uses separate object of collection variables. This makes every request in UI independent.

### ISSUE THREAD

**Issue by mirkogolze** [usebruno/bruno#188]
When I run a folder from the UI then the collection variables are used for all requests in the folder. But I don't see the responses.
When I have requests belonging to one to another where one request stores a collection variable, then the next request can't use the value of the stored variable because running it as single request makes them independent.
So I can't test the flow of my collection by running the request one by one and see the result.

With additionally added scripts this is not so easy too, because I can't use `fs` to work with temporary files or so. I only could use databases. But that would be like shooting sparrows with cannons.

**Comment by helloanoop** [comment]
@mirkogolze 
Can you elaborate the requirement a bit more.?

> When I run a folder from the UI then the collection variables are used for all requests in the folder. But I don't see the responses.

If you click on "200 ok" response status, you should see the response
![image](https://github.com/usebruno/bruno/assets/9350733/649be7c9-051d-46cc-9996-d4650e85396d)

> where one request stores a collection variable, then the next request can't use the value of the stored variable because running it as single request makes them independent.

The collection variables are a the collection scope. If a requests sets it using `bru.setVar(name, value)` then any subsequent request can access it using `bru.getVar(name)`

Can you describe your high level use case? I might be able to suggest some simpler approach.

**Comment by mirkogolze** [comment]
SampleCollection: [VarTesting.zip](https://github.com/usebruno/bruno/files/12595876/VarTesting.zip)

SetVar
```
test("should return the IP", function() {
  const data = res.getBody();
  expect(data.ip).to.be.a('string');
  bru.setVar('myIP', data.ip);
});
```
GetVar
```
test("should get the IP from var", function() {
  const ip = bru.getVar('myIP');
  expect(ip).to.be.a('string');
});
```
Running the folder in the UI
![image](https://github.com/usebruno/bruno/assets/66121941/09db3c9b-aea3-4152-a627-fef9ba63570e)
All is fine!

Running the request one by one in the UI
![image](https://github.com/usebruno/bruno/assets/66121941/2c5bdb46-2d89-4ccb-8daa-e2e23f5d95ba)
Variable is saved

![image](https://github.com/usebruno/bruno/assets/66121941/c1bbd2f9-b899-4de9-87a7-1a555c9e1406)
Variable should be read from Collection - but collection variables are empty.

I think now the requests are running with the method `runSingleRequest`, where the collectionVariables are freshly instantiated.
To implement a flow of requests, I wan't to run the requests one by one, see the responses, implement my tests and assertions. 
Read data from responses, store to collection variables and use that data in pre-request-scripts.
This is not possible because every request has its own collection variables. So we would need a kind of context inside the UI on scope of the collection.

**Comment by helloanoop** [comment]
Thanks @mirkogolze for the detailed writeup of the issue.
I have fixed this and this is now available in the latest v0.14.0 release

PS: In this release, I've also added support for printing console.log() statements (inside the scripts/tests) in the devtools for easier debugging.

**Comment by mirkogolze** [comment]
Thanks. Works now as expected.
